hbase-server/src/main/java/org/apache/hadoop/hbase/client/ClientSideRegionScanner.java

```diff
@@ -60,6 +62,14 @@ public ClientSideRegionScanner(Configuration conf, FileSystem fs,
     region = HRegion.newHRegion(CommonFSUtils.getTableDir(rootDir, htd.getTableName()), null, fs,
       conf, hri, htd, null);
     region.setRestoredRegion(true);
+    // non RS process does not have a block cache, and this a client side scanner,
+    // create one for MapReduce jobs to cache the INDEX block
+    conf.setFloat(HConstants.HFILE_BLOCK_CACHE_SIZE_KEY,
+      conf.getFloat(HConstants.HBASE_CLIENT_SCANNER_BLOCK_CACHE_SIZE_KEY,
+        HConstants.HBASE_CLIENT_SCANNER_BLOCK_CACHE_SIZE_DEFAULT));
```

Comment:
[DISCUSS] we can discuss if we should just set to `HFILE_BLOCK_CACHE_SIZE_KEY` to `0.1f` , or we just don't set it

Reply:
I guess we could actually do this by doing the opposite of the `MemorySizeUtil.getOnHeapCacheSize` method.

1. Take a discrete value from the user (32MB)
2. Compute 32MB / MemorySizeUtil.safeGetHeapMemoryUsage()
3. Set the value from step 2 as the bucket cache size

Reply:
introduced a new key `hfile.block.cache.fixed.size` and set to it whenever it's provided. I thought about comparing which one is smaller, however, I chose to keep it simpler because it's already complicated enough in the user experience between `hfile.block.cache.fixed.size` and `hfile.block.cache.size`